lib/rubygems/core_ext/kernel_require.rb

```diff
@@ -49,9 +49,12 @@ def require path
     # If there are no unresolved deps, then we can use just try
     # normal require handle loading a gem from the rescue below.
 
-    if Gem::Specification.unresolved_deps.empty? then
+    begin
       RUBYGEMS_ACTIVATION_MONITOR.exit
-      return gem_original_require(path)
+      original = gem_original_require(path)
+      return original
+    rescue LoadError
```

Comment:
The only thing I can think of is re-implementing ruby core’s require inside rubygems, which is obviously undesireable. (I’m presuming there’s no version of require that simply doesn’t throw exceptions that I don’t know about?)
> On May 20, 2015, at 2:40 PM, Aaron Patterson <notifications@github.com> wrote:
> 
> In lib/rubygems/core_ext/kernel_require.rb <https://github.com/rubygems/rubygems/pull/1252#discussion_r30740658>:
> 
> >        RUBYGEMS_ACTIVATION_MONITOR.exit
> > -      return gem_original_require(path)
> > +      original = gem_original_require(path)
> > +      return original
> > +    rescue LoadError
> Isn't this going to cause many exceptions during require? Seems like it would make -d very noisy. Is there another way to fix this?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/rubygems/rubygems/pull/1252/files#r30740658>.
>